### ISSUE TITLE
SpaceAroundMapEntryColon rule no longer works with characterAfterColonRegex

### ISSUE THREAD

**Issue by nre-ableton** [CodeNarc/CodeNarc#502]
I'm pretty sure that this broke with https://github.com/CodeNarc/CodeNarc/commit/bd78f989438ff4ff538cd82984660c24fb902479 (see https://github.com/CodeNarc/CodeNarc/issues/430), but looking at the diff, I'm not entirely sure _how_ this could happen.

Regardless, since CodeNarc 1.5, the `SpaceAroundMapEntryColon` is throwing all sorts of errors like so:

> `SpaceAroundMapEntryColon: The colon for the literal Map entry for key [ignoreMethodNames] within class None is not followed by a match for regular expression [\\s]`

My (abbreviated) `ruleset.groovy` looks like this:

```groovy
  ruleset('rulesets/formatting.xml') {
    SpaceAroundMapEntryColon(characterAfterColonRegex: /\\s/)
  }
```

I have also tried `SpaceAroundMapEntryColon(characterAfterColonRegex: /\s/)` and even `SpaceAroundMapEntryColon(characterAfterColonRegex: / /)` (with a literal space), but the result is the same in all cases.

**Comment by chrismair** [comment]
Can you provide some example Groovy code that causes these incorrect violations?

I'm pretty sure you want `/\s/` as the regex.

**Comment by nre-ableton** [comment]
Sure, the [Jenkinsfile of groovylint](https://github.com/AbletonDevTools/groovylint/blob/master/Jenkinsfile) is one such example. Here's an excerpt:

```groovy
devToolsProject.run(
  setup: { data ->
    data['venv'] = virtualenv.create('python3.8')
    data.venv.run('pip install -r requirements-dev.txt -r requirements.txt')
  },
  // etc...
)
```

For the above snippet, CodeNarc produces these errors:

> `ERROR ./Jenkinsfile:15: SpaceAroundMapEntryColon: The colon for the literal Map entry for key [setup] within class None is not preceded by a match for regular expression [\S]`
> `ERROR ./Jenkinsfile:15: SpaceAroundMapEntryColon: The colon for the literal Map entry for key [setup] within class None is not followed by a match for regular expression [\s]`

**Comment by chrismair** [comment]
Are you perhaps using Groovy 3.0 or later? I added a test for the above code and it produced no violations when running Groovy 2.4, but produced the above errors running Groovy 3.0.3.

I am working another issue #495 to fix Groovy 3.0 issues. I intend to add a test for the above sample code to the test class so that I can verify it gets fixed.

**Comment by nre-ableton** [comment]
I'm pretty sure we are using Groovy 2.4.12 (the same version Jenkins ships with), but I can double check this.

**Comment by nre-ableton** [comment]
Ah, it turns out that I was using Groovy 2.4.12 while in the Docker container, but the system Groovy (so version 3.0.3) when running locally. So yes, this is a Groovy 3.x issue.

**Comment by chrismair** [comment]
Okay good. Then I am closing as fixed by #495. Thanks for checking.

**Comment by nre-ableton** [comment]
Great, thanks for the fix @chrismair
